### ISSUE TITLE
issue with assert_queued_in incorrect comparison

### ISSUE THREAD

**Issue by epinault** [justinweiss/resque_unit#43]
I am using rails 4.2.4, Ruby 2.1.5 and your gem 0.4.8. I am migrating from 4.0 and everything worked ok on 4.0 and same ruby version. Something may have changed in Rails that affects your code

I am getting some strange error. Same test sometime passes , sometime i get

Minitest::Assertion:         Notifications::NotificationWorker should have been queued in notify_app before 2015-09-11 14:44:36 -0700: ["{\"class\":\"Notifications::NotificationWorker\",\"args\":[1],\"timestamp\":\"2015-09-11T14:44:36.012-07:00\"}"].

Noticed the time is the same beside one include the microseconds? Time Resolution seems to include in ruby 2.1.5 and Rails 4.2 the microseconds. Make the test fails .... 

Noticed that with irb

```
% irb
>> Time.now
=> 2015-09-11 14:47:35 -0700
```

Here is the same with Rails console
 pry(main)> Time.now

```
=> "2015-09-11T14:48:04.720-07:00"
```

Because of that difference , cannot get the test to pass correctly all the time.

**Comment by epinault** [comment]
I believe the difference is between inspect vs to_s . 

in the rails console

[7] pry(main)> Time.now.to_s
=> "2015-09-11 14:59:00 -0700"
[8] pry(main)> Time.now
=> "2015-09-11T14:59:07.348-07:00"

**Comment by epinault** [comment]
I realize this might also sometime be due to a delay to find them in redis on my vm too

**Comment by epinault** [comment]
Hum. Looking at it deeper the problem might be also somewhere else too

  def matching_jobs(queue, klass, args = nil)
    normalized_args = Resque.decode(Resque.encode(args)) if args
    queue.select {|e| e["class"] == klass.to_s && (!args || e["args"] == normalized_args )}
  end

In the assert queued in 

Resque.all(queue_name)  => [{"class"=>"Notifications::NotificationWorker", "args"=>[1], "timestamp"=>"2015-09-11T15:37:02.023-07:00"}]
matching_jobs(Resque.all(queue_name) => []

I looked into   

def matching_jobs(queue, klass, args = nil)
    normalized_args = Resque.decode(Resque.encode(args)) if args
    p normalized_args  => [9]
end

Not sure what is going on but sometime the arg comes with the wrong value

**Comment by justinweiss** [comment]
Hm, that's weird, but not too surprising. I'll take a look when I have a chance. Thanks for letting me know!

**Comment by justinweiss** [comment]
I just released a beta that should fix the issue -- if you want, try `v1.0.0.beta.1`.

**Comment by epinault** [comment]
cool! thanks!
